packages/web-client/src/webdav/listFiles.ts

```diff
@@ -20,7 +20,7 @@ export const ListFilesFactory = ({ sdk }: WebDavOptions) => {
       space: SpaceResource,
       { path, fileId }: { path?: string; fileId?: string | number } = {},
       { depth = 1, davProperties }: ListFilesOptions = {}
-    ): Promise<Resource[]> {
+    ): Promise<{ folder: Resource; children: Resource[] }> {
```

Comment:
- Please define an interface for the return type
- `children` is optional, so you need to declare it as `children?: Resource[]`
- `folder` is not such a good fit, naming wise, since it can also be a space. Maybe `node` and `children` works better? @dschmidt what's your opinion on this?

Reply:
Naming is hard... 

folder is very bad, I agree. I'd say even file would be better than folder, as every folder is a file but not every file a folder (in POSIX terms). node is a bit weird imho as we don't use that term anywhere afaict. 

I know it's semantically weak, but what about resource? main? mainResource? result? 
At least it's not more specific than the actual result - which might be misleading (eg when the result is a space node seems a bit off)

und jetzt weiß ich auch nicht 😅

Reply:
i also think node is a better fit 👍🏼

Reply:
alright

Reply:
fixed 👍🏼